daemon/SecretsImpl/secrets.cpp

```diff
@@ -1814,14 +1822,60 @@ void Daemon::ApiImpl::SecretsRequestQueue::handleFinishedRequest(
                 qCWarning(lcSailfishSecretsDaemon) << "GetStandaloneSecretRequest:" << request->requestId << "finished as pending!";
                 *completed = true;
             } else {
-                QByteArray secret = request->outParams.size()
-                        ? request->outParams.takeFirst().toByteArray()
-                        : QByteArray();
+                Secret secret = request->outParams.size()
+                        ? request->outParams.takeFirst().value<Secret>()
+                        : Secret();
                 if (request->isSecretsCryptoRequest) {
-                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << QVariant::fromValue<QByteArray>(secret));
+                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << QVariant::fromValue<Secret>(secret));
                 } else {
                     request->connection.send(request->message.createReply() << QVariant::fromValue<Result>(result)
-                                                                            << QVariant::fromValue<QByteArray>(secret));
+                                                                            << QVariant::fromValue<Secret>(secret));
+                }
+                *completed = true;
+            }
+            break;
+        }
+        case FindCollectionSecretsRequest: {
+            Result result = request->outParams.size()
+                    ? request->outParams.takeFirst().value<Result>()
+                    : Result(Result::UnknownError,
+                             QLatin1String("Unable to determine result of FindCollectionSecretsRequest request"));
+            if (result.code() == Result::Pending) {
+                // shouldn't happen!
+                qCWarning(lcSailfishSecretsDaemon) << "FindCollectionSecretsRequest:" << request->requestId << "finished as pending!";
+                *completed = true;
+            } else {
+                QVector<Secret::Identifier> identifiers = request->outParams.size()
+                        ? request->outParams.takeFirst().value<QVector<Secret::Identifier> >()
+                        : QVector<Secret::Identifier>();
+                if (request->isSecretsCryptoRequest) {
+                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << QVariant::fromValue<QVector<Secret::Identifier> >(identifiers));
+                } else {
+                    request->connection.send(request->message.createReply() << QVariant::fromValue<Result>(result)
+                                                                            << QVariant::fromValue<QVector<Secret::Identifier> >(identifiers));
```

Comment:
This coding style is hard to follow.  
Suggestion:

```cpp
auto reply = request->message.createReply();
reply << QVariant::fromValue(result);
reply << QVariant::fromValue(identifiers);
request->connection.send(reply);
```

This is equivalent, but more readable.

Reply:
Why is what I've done harder to read?

Reply:
At least because of the width of the code. You clearly see what is happening at a glance, without following the text and the indentation.

Reply:
This is, again, of course, a matter of opinion. Perhaps we just come from a different background.

To my eyes, the kind of code where each statement fulfills only one clearly understandable purpose is easier to read than a bunch of things mixed into a single statement. As @vityafx says it's easier to glance what's happening.

Also, when you are debugging the code and wish to know what is passed to `request->connection.send()`, then in the simpler case you can just place a breakpoint and then see it visualized in the debugger. Whereas if it is written as a single statement, it is a bit harder to debug.

Like I said, these are just my 20 cents.

Reply:
We don't have an 80-column restriction, and IMO for good reason.  Indentation is an important part of code style, indeed there are languages like python which rely solely on it to convey and enforce expression block semantics.  But I agree that some styles of indentation are more readable than others, and you may very well be correct that this style is less readable than the one you suggest.

Regarding splitting over multiple lines, it is somewhat less atomic (e.g. there exists potential for a separate, unrelated instruction with side-effects to be inserted there; alternatively, implies that the variable might be used elsewhere), whereas the syntax I use makes it syntactically clear that the thing being created is solely for the argument and should have no side effects.

Regardless, you may very well be right.  Nonetheless, I would not change this single instance (or the multiple - but still a tiny subset of the total - instances) in this PR because then we would be introducing style inconsistencies with the rest of the code base (which IMO is even worse for maintainability). IMO the correct way to resolve this, if we agree that it should be changed, would be to change such style across the entire project, in a single, atomic commit / PR.  As I have said many times: your PRs are welcome.  From OMP or from Venemo or from community contributors.